test/test_filesystem_helper_std.cpp

```diff
@@ -0,0 +1,520 @@
+    }
+    {
+      auto p = path("");
+      EXPECT_FALSE(p.is_absolute());
+    }
+  } else {
+    {
+      auto p = path("/foo/bar/baz");
+      EXPECT_TRUE(p.is_absolute());
+    }
+    {
+      auto p = path("foo/bar/baz");
+      EXPECT_FALSE(p.is_absolute());
+    }
+    {
+      auto p = path("f");
+      EXPECT_FALSE(p.is_absolute());
+    }
+    {
+      auto p = path("");
+      EXPECT_FALSE(p.is_absolute());
+    }
+  }
+}
+
+TEST(TestFilesystemHelper, correct_extension)
+{
+  {
+    auto p = path(build_extension_path());
+    auto ext = p.extension();
+    EXPECT_EQ(".yml", ext.string());
+  }
+  {
+    auto p = path(build_double_extension_path());
+    auto ext = p.extension();
+    EXPECT_EQ(".yml", ext.string());
+  }
+  {
+    auto p = path(build_no_extension_path());
+    auto ext = p.extension();
+    EXPECT_EQ("", ext.string());
+  }
+}
+
+TEST(TestFilesystemHelper, is_empty)
+{
+  auto p_no_arg = path();
+  EXPECT_TRUE(p_no_arg.empty());
+
+  auto p_empty_string = path("");
+  EXPECT_TRUE(p_empty_string.empty());
+}
+
+TEST(TestFilesystemHelper, exists)
+{
+  {
+    auto p = path("");
+    EXPECT_FALSE(std::filesystem::exists(p));
+  }
+  {
+    auto p = path(".");
+    EXPECT_TRUE(std::filesystem::exists(p));
+  }
+  {
+    auto p = path("..");
+    EXPECT_TRUE(std::filesystem::exists(p));
+  }
+  {
+    if (is_win32) {
+      auto p = path("\\");
+      EXPECT_TRUE(std::filesystem::exists(p));
+    } else {
+      auto p = path("/");
+      EXPECT_TRUE(std::filesystem::exists(p));
+    }
+  }
+}
+
+/**
+ * Test filesystem manipulation API.
+ *
+ * NOTE: expects the current directory to be write-only, else test will fail.
+ *
+ */
+TEST(TestFilesystemHelper, filesystem_manipulation)
+{
+  auto dir = path(build_directory_path());
+  (void)std::filesystem::remove_all(dir);
+  EXPECT_FALSE(std::filesystem::exists(dir));
+  EXPECT_TRUE(std::filesystem::create_directories(dir));
+  EXPECT_TRUE(std::filesystem::exists(dir));
+  EXPECT_TRUE(std::filesystem::is_directory(dir));
+  EXPECT_FALSE(std::filesystem::is_regular_file(dir));
+
+  auto file = dir / "test_file.txt";
+  uint64_t expected_file_size = 0;
+  {
+    std::ofstream output_buffer{file.string()};
+    output_buffer << "test";
+    expected_file_size = static_cast<uint64_t>(output_buffer.tellp());
+  }
+
+  EXPECT_TRUE(std::filesystem::exists(file));
+  EXPECT_TRUE(std::filesystem::is_regular_file(file));
+  EXPECT_FALSE(std::filesystem::is_directory(file));
+  std::error_code ec;
+  EXPECT_FALSE(std::filesystem::create_directories(file, ec));
+  EXPECT_GE(std::filesystem::file_size(file), expected_file_size);
+  EXPECT_THROW((void)std::filesystem::file_size(dir), std::filesystem::filesystem_error) <<
+    "file_size is only applicable for files!";
+  EXPECT_FALSE(std::filesystem::remove(dir, ec));
+  EXPECT_TRUE(std::filesystem::remove(file, ec));
+  EXPECT_THROW((void)std::filesystem::file_size(file), std::filesystem::filesystem_error);
```

Comment:
```suggestion
  EXPECT_THROW(std::filesystem::file_size(file), std::filesystem::filesystem_error);
```

Reply:
removing this, it will generate a warning because we are not using the result of the call

Reply:
Oh, right.  We have to leave this one (and the other `file_size` one) in place, sorry about that.